### PR TITLE
Create angular service to manage time tracking data

### PR DESCRIPTION
**Pull request by jrbotros**
Ripped out of my larger branch for the sake of smaller PRs! I'll be submitting two more PRs for the `time-tracking` branch, and then we can merge the entire branch in when we're ready.

I've manually tested things in the console and everything seems okay (can't wait for front-end testing), and when I PR the interfaces we can be sure!

### CODE REVIEW

File: orchestra/static/orchestra/timing/service.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+(function() {
+  'use strict';
+
+  angular.module('orchestra.timing')
+    .factory('timingService', function($http, TimeEntry) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js]: would call it timeEntryService for clarity

File: orchestra/static/orchestra/timing/service.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          $http.get(service.apiUrl, {
+            params: {
+              min_date: minDate.format('YYYY-MM-DD'),
+              max_date: maxDate.format('YYYY-MM-DD'),
+            }
+          })
+          .then(function(response) {
+            // Initialize dates in range
+            var numDays = maxDate.diff(minDate, 'days');
+            service.entriesByDate = {};
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js:33]: should this be declared at the top with entries?

File: orchestra/static/orchestra/timing/service.js (hunk 1 above)
Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js]: oh i see you have another service called TimeEntry ha

File: orchestra/static/orchestra/timing/time-entry.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        var entry = this;
+
+        var createUrl = '/orchestra/api/interface/time_entries/';
+        this.date = date || moment();
+        $http.post(createUrl, {
+          date: date.format('YYYY-MM-DD'),
+          time_worked: moment.duration().stamp(),
+
+          // Add fake timer start time to keep ordered correctly
+          timer_start_time: moment().toISOString()
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/time-entry.js]: can you order by id instead? Concerned that having this timer_start_time in the db would be confusing.

File: orchestra/static/orchestra/timing/time-entry.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        })
+        .then(function(response) {
+          entry.initWithData(response.data);
+          entry.editing = true;
+        }, function(entry) {
+          alert('Could not add new time entry.');
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/time-entry.js:94]: Is using alert the standard for error modals in orchestra? I'm also wondering for something I'm working on.

File: orchestra/static/orchestra/timing/time-entry.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      /**
+       * Updates TimeEntry data on the server.
+       */
+      TimeEntry.prototype.save = function() {
+        var entry = this;
+
+        var data = angular.copy(entry);
+
+        // Convert datetimes to moments for easy manipulation
+        var datetimeFields = ['timer_start_time', 'timer_stop_time'];
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/time-entry.js:117]: Oh, I also realized that I removed timer_start_time/timer_stop_time from the TimeEntry serializer. have you rebased on master recently?

File: orchestra/static/orchestra/timing/service.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+          });
+        },
+        addEntry: function(data) {
+          var entry = new TimeEntry(data);
+          this.entries.push(entry);
+          this.addEntryToDate(entry, entry.date);
+          return entry;
+        },
+        deleteEntry: function(entry) {
+          if (confirm('Are you sure you want to delete this entry?')) {
```
Comment (lydiagu) [inline orchestra/static/orchestra/timing/service.js:76]: is confirm() the standard in orchestra for confirmation boxes?

**Comment by lydiagu** [comment]
Thanks for making a small PR!! 🎉 

Done reviewing

File: orchestra/static/orchestra/timing/time-entry.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/time-entry.js]: How do you feel about a `created_at` field instead?

File: orchestra/static/orchestra/timing/time-entry.js (hunk 4 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/time-entry.js:94]: Right now, yes (it's terrible, I know)

File: orchestra/static/orchestra/timing/service.js (hunk 6 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/service.js:76]: Yep, no replacement yet

File: orchestra/static/orchestra/timing/service.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/service.js]: Maybe just `timeEntries`?

File: orchestra/static/orchestra/timing/service.js (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/service.js:33]: Doesn't make a ton of difference, but probably more clear!

File: orchestra/static/orchestra/timing/time-entry.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/time-entry.js]: Talked about offline, will order by `id` for now!

File: orchestra/static/orchestra/timing/time-entry.js (hunk 5 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/time-entry.js:117]: This was just getting ignored by the serializer and I didn't realize it—will take out now.

File: orchestra/static/orchestra/timing/service.js (hunk 1 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/service.js]: ✅

File: orchestra/static/orchestra/timing/service.js (hunk 2 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/service.js:33]: ✅

File: orchestra/static/orchestra/timing/time-entry.js (hunk 3 above)
Comment (jrbotros) [inline orchestra/static/orchestra/timing/time-entry.js]: ✅

**Pull request merged**